### PR TITLE
Create AdjMatrix.java

### PR DESCRIPTION
**Pull request by jaycoug**

### CODE REVIEW

**Review by dwilliams56 — APPROVED** [review]

File: src/main/Boards/AdjMatrix.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	  }
+
+	  public AdjMatrix mult(AdjMatrix m2) {
+	    if (m2.size() == this.size()) {
+	      Matrix retMatrix = new UpperSPDPackMatrix(size);
+	      data.mult(m2.getData(), retMatrix);
+	      return new AdjMatrix(retMatrix);
+	    } else
+	      return null;
+	  }
```
Comment (djimenezmsu) [inline src/main/Boards/AdjMatrix.java:44]: Just a suggestion, but for debugging purposes, I think we should throw an invalid operation exception with a message referencing where the exception came from. That way, if we run into a null reference exception somewhere leading here, we'll be able to quickly diagnose the issue.

File: src/main/Boards/AdjMatrix.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+	  public void write(int a, int b, int value) {
+	    if (a == b)
+	      value = LINK;
+	    internalWrite(a, b, value);
+	  }
+
+	  public int read(int x, int y) {
+	    return (int) data.get(x, y);
+	  }
```
Comment (djimenezmsu) [inline src/main/Boards/AdjMatrix.java:54]: For type safety, we should use Integer.tryParse(data.get(x, y)) and wrap it in a try catch. In the catch we should throw the exception up with a message about where it came from.

**Review by djimenezmsu — CHANGES_REQUESTED** [review]
see above

**Review by mrichardson3065 — APPROVED** [review]

**Review by djimenezmsu — APPROVED** [review]
I've went it and just added the exception thrown with a message of where it came from. I tried adding a safer type casting of the read method but I couldn't figure out a way that wasn't worse than it is. Let me know if this is okay

**Pull request merged**